drivers/sensor/lis2dh/lis2dh.c

```diff
@@ -207,17 +410,69 @@ static int lis2dh_acc_range_set(const struct device *dev, int32_t range)
 }
 #endif
 
+int lis2dh_acc_act_config(const struct device *dev,
+			    enum sensor_attribute attr,
+			    const struct sensor_value *val)
+{
+	struct lis2dh_data *lis2dh = dev->data;
+	int ret;
+
+	if ((uint16_t)attr == SENSOR_ATTR_LIS2DH_ACT_TH) {
+		uint8_t range_g, reg_val;
+		uint32_t act_th_ums2;
+
+		ret = lis2dh->hw_tf->read_reg(dev, LIS2DH_REG_CTRL4,
+						 &reg_val);
+		if (ret < 0) {
+			return ret;
+		}
+
+		/* fs reg value is in the range 0 (2g) - 3 (16g) */
+		range_g = 2 * (1 << ((LIS2DH_FS_MASK & reg_val)
+				      >> LIS2DH_FS_SHIFT));
+
+		act_th_ums2 = val->val1 * 1000000 + val->val2;
+
+		/* make sure the provided threshold does not exceed range */
+		if ((act_th_ums2 - 1) > (range_g * SENSOR_G)) {
+			return -EINVAL;
+		}
+
+		/* 7 bit full range value */
+		reg_val = 128 / range_g * (act_th_ums2 - 1) / SENSOR_G;
+
+		LOG_INF("act_ths=0x%x range_g=%d ums2=%u", reg_val,
+			    range_g, act_th_ums2 - 1);
+
+		return lis2dh->hw_tf->write_reg(dev, LIS2DH_REG_ACT_THS,
+						  reg_val);
+	} else { /* SENSOR_ATTR_LIS2DH_ACT_DUR */
+		/*
+		 * slope duration is measured in number of samples:
+		 * N/ODR where N is the register value
+		 */
+		if (val->val1 < 0 || val->val1 > 127) {
+			return -ENOTSUP;
+		}
+
+		LOG_INF("act_dur=0x%x", val->val1);
+
+		return lis2dh->hw_tf->write_reg(dev, LIS2DH_REG_ACT_DUR,
+						  val->val1);
+	}
+}
+
 static int lis2dh_acc_config(const struct device *dev,
 			     enum sensor_channel chan,
 			     enum sensor_attribute attr,
 			     const struct sensor_value *val)
 {
-	switch (attr) {
-#ifdef CONFIG_LIS2DH_ACCEL_RANGE_RUNTIME
+	switch ((uint16_t)attr) {
```

Comment:
Is there a reason you are casting?  The enum is by default an int value

Reply:
This has to do with adding custom sensor attributes. If the value is not casted it cannot be compared to two different enumerators

Reply:
Then my advice would be not to use 2 different enumerators as this can lead to a lot of confusion. Try to place every in one enum. Having an enum helps the compiler check which cases there should be